internal/ingress/controller/hasher.go

```diff
@@ -1,21 +1,387 @@
+
+	writeCertificateAuth(s.CertificateAuth, hash)
+
+	writeString(s.ServerSnippet, hash)
+
+	writeString(s.SSLCiphers, hash)
+
+	writeString(s.AuthTLSError, hash)
+
+	for _, l := range s.Locations {
+		writeLocation(*l, hash)
+	}
+}
+
+func writeCertificateAuth(s authtls.Config, hash io.Writer) {
+	writeString(s.Secret, hash)
+	writeString(s.CAFileName, hash)
+	writeString(s.PemSHA, hash)
+	writeString(s.VerifyClient, hash)
+	writeString(s.ErrorPage, hash)
+	writeString(s.AuthTLSError, hash)
+	writeString(s.AuthSSLCert.CAFileName, hash)
+	writeString(s.AuthSSLCert.PemSHA, hash)
+	writeString(s.AuthSSLCert.Secret, hash)
+	writeInt(s.ValidationDepth, hash)
+	writeBool(s.PassCertToUpstream, hash)
+}
+
+func writeCorsConfig(c cors.Config, hash io.Writer) {
+	writeString(c.CorsAllowOrigin, hash)
+	writeBool(c.CorsEnabled, hash)
+	writeString(c.CorsAllowMethods, hash)
+	writeString(c.CorsAllowHeaders, hash)
+	writeBool(c.CorsAllowCredentials, hash)
+	writeInt(c.CorsMaxAge, hash)
+}
+
+func writeExternalAuth(l authreq.Config, hash io.Writer) {
+	writeString(l.URL, hash)
+	writeString(l.Host, hash)
+	writeString(l.SigninURL, hash)
+	writeString(l.Method, hash)
+	writeString(l.RequestRedirect, hash)
+	writeString(l.AuthSnippet, hash)
+	for _, head := range l.ResponseHeaders {
+		writeString(head, hash)
+	}
+}
+
+func writeRateLimit(r ratelimit.Config, hash io.Writer) {
+	writeString(r.Name, hash)
+	writeString(r.ID, hash)
+	writeInt(r.LimitRate, hash)
+	writeInt(r.LimitRateAfter, hash)
+	writeZone(r.Connections, hash)
+	writeZone(r.RPS, hash)
+	writeZone(r.RPM, hash)
+	for _, i := range r.Whitelist {
+		writeString(i, hash)
+	}
+}
+
+func writeZone(z ratelimit.Zone, hash io.Writer) {
+	writeString(z.Name, hash)
+	writeInt(z.Limit, hash)
+	writeInt(z.SharedSize, hash)
+	writeInt(z.Burst, hash)
+}
+
+func writeProxy(p proxy.Config, hash io.Writer) {
+	writeString(p.BodySize, hash)
+	writeString(p.BufferSize, hash)
+	writeString(p.CookieDomain, hash)
+	writeString(p.CookiePath, hash)
+	writeString(p.NextUpstream, hash)
+	writeString(p.ProxyRedirectFrom, hash)
+	writeString(p.ProxyRedirectTo, hash)
+	writeString(p.RequestBuffering, hash)
+	writeString(p.ProxyBuffering, hash)
+	writeInt(p.ConnectTimeout, hash)
+	writeInt(p.SendTimeout, hash)
+	writeInt(p.ReadTimeout, hash)
+	writeInt(p.BuffersNumber, hash)
+	writeInt(p.NextUpstreamTries, hash)
+}
+
+func writeAuthTLSConfig(a auth.Config, hash io.Writer) {
+	writeString(a.Type, hash)
+	writeString(a.Realm, hash)
+	writeString(a.File, hash)
+	writeString(a.FileSHA, hash)
+	writeString(a.Secret, hash)
+	writeBool(a.Secured, hash)
+}
+
+func writeLocation(l ingress.Location, hash io.Writer) {
+	writeString(l.Path, hash)
+
+	writeBool(l.IsDefBackend, hash)
+
+	writeString(l.Backend, hash)
+
+	writeString(l.Service.GetNamespace(), hash)
+	writeString(l.Service.GetName(), hash)
+
+	writeString(l.Port.StrVal, hash)
+
+	writeAuthTLSConfig(l.BasicDigestAuth, hash)
+
+	writeString(*l.Denied, hash)
```

Comment:
This can segfault if Denied (pointer to string) is null (as I saw in testing earlier). The fix I implemented is in our debug branch.